.ci/validate-examples.py

```diff
@@ -0,0 +1,205 @@
+        if len(notebook_files) == 0:
+            msg = f"No notebooks were found in '{full_dir}' or its subdirectories"
+            ERRORS.add(msg)
+        else:
+            for notebook_file in notebook_files:
+                with open(notebook_file, "r") as f:
+                    notebook_dict = json.loads(f.read())
+                non_empty_cells = 0
+                for cell in notebook_dict["cells"]:
+                    if cell.get("outputs", []) or cell.get("execution_count", None):
+                        non_empty_cells += 1
+                if non_empty_cells > 0:
+                    msg = (
+                        f"Found {non_empty_cells} non-empty cells in '{notebook_file}'. "
+                        "Clear all outputs and re-commit this file."
+                    )
+                    ERRORS.add(msg)
+
+        # all files and directories should have appropriate name
+        for fname in all_files:
+            if os.path.isdir(fname):
+                base_name = os.path.basename(os.path.normpath(fname))
+                matches_name_pattern = bool(re.search(DIRECTORY_REGEX, base_name))
+                msg = (
+                    f"[here] All directories should be named with only "
+                    "lower alphanumeric characters and dashes. "
+                    f"'{fname}` violates this rule."
+                )
+            else:
+                base_name = os.path.basename(fname)
+                matches_name_pattern = bool(re.search(FILENAME_REGEX, base_name))
+                msg = (
+                    f"All files should be named with only "
+                    "alphanumeric characters, dashes, and periods. "
+                    f"'{fname}` violates this rule."
+                )
+            if not matches_name_pattern:
+                ERRORS.add(msg)
+
+        # each example must have a README.md
+        readme_file = os.path.join(full_dir, "README.md")
+        if not os.path.isfile(readme_file):
+            msg = f"Every example must have a README.md. '{full_dir}' does not."
+            ERRORS.add(msg)
+
+        # each example must have a .saturn/ folder
+        saturn_dir = os.path.join(full_dir, SATURN_DIR_NAME)
+        if not os.path.isdir(saturn_dir):
+            msg = f"'{full_dir}' does not include a '{SATURN_DIR_NAME}/' directory"
+            ERRORS.add(msg)
+
+        saturn_json = os.path.join(saturn_dir, SATURN_JSON_NAME)
+        if not os.path.isfile(saturn_json):
+            msg = f"Did not find saturn.json in '{saturn_dir}'. This file is required."
+            ERRORS.add(msg)
+            continue
+
+        with open(saturn_json, "r") as f:
+            saturn_config = json.loads(f.read())
+
+        for required_key in SATURN_JSON_KEYS:
+            if required_key not in saturn_config.keys():
+                msg = f"'{saturn_json}' missing required key: '{required_key}'"
+                ERRORS.add(msg)
+
+        project_image = saturn_config["image"]
```

Comment:
should there be some validations for `jupyter` also?

Reply:
definitely! Is it ok if I do that in a follow-up PR after this one? I want to take this small steps at a time.